hcl2template/parser.go

```diff
@@ -147,16 +140,38 @@ func (p *Parser) Parse(filename string, varFiles []string, argVars map[string]st
 	}
 
 	// Before we go further, we'll check to make sure this version can read
-	// that file, so we can produce a version-related error message rather than
+	// all files, so we can produce a version-related error message rather than
 	// potentially-confusing downstream errors.
 	versionDiags := cfg.CheckCoreVersionRequirements(p.CorePackerVersion)
 	diags = append(diags, versionDiags...)
 	if versionDiags.HasErrors() {
 		return cfg, diags
 	}
 
+	// Decode required_plugins blocks and create implicit required_plugins
+	// blocks. Implicit required_plugins blocks happen when a builder or another
+	// plugin cannot be found, for example if one uses :
+	//  source "amazon-ebs" "example" { ... }
+	// And no `amazon-ebs` builder can be found. This will then be the
+	// equivalent of having :
+	//  packer {
+	//    required_plugins {
+	//      amazon = "latest"
+	//    }
+	//  }
+	// Note: using `latest` ( or actually an empty string ) in a config file
+	// does not work and packer will ask you to pick a version
+	{
+		for _, file := range files {
+			diags = append(diags, cfg.decodeRequiredPluginsBlock(file)...)
+		}
+		for _, file := range files {
+			diags = append(diags, cfg.decodeImplicitRequiredPluginsBlocks(file)...)
```

Comment:
This only works for the plugins we maintain, right?

Reply:
Well, I think that this 'guesser' should have as little info as possible and cover all already 'incore' plugins.

Reply:
( nothing works there, this is mainly TODO, I just wanted to put the wiring there for later. )